### ISSUE TITLE
App is blocked by Google 

### ISSUE THREAD

**Issue by hoangsonqn** [ericyd/gdrive-copy#193]
### Oath of good faith

* [ ] I, the author of this Issue, do solemnly swear that I've looked _real hard_ for duplicate issues and I came up empty!!

**Comment by gmendezg** [comment]
Same problem.

https://stackoverflow.com/questions/66633659/can-not-run-trivial-app-script-bound-to-new-blank-document-get-this-app-is-bl

Related?

**Comment by ycarogabriel** [comment]
Same issue here

**Comment by santi-ifontana** [comment]
Hey everyone, same issue here. Probably you are using Google Workspace, right?

Do the following to fix it (only valid for Google Workspace accounts).

1. Go here: https://admin.google.com/ac/owl/list?tab=configuredApps
2. Be sure to log in with an Admin user and the same console where the user trying to do the copy lives.
3. Click "Configure new app".
4. On the search bar that appears, type this: 983475130545-e81q4d30k615fsr942krcejaajriv23a.apps.googleusercontent.com
5. Copy folder should be found.
6. Select Trust and save.
7. Try again in a few minutes (sometimes it takes some time to propagate).
8. Good luck!
9. Please confirm if it worked for you!

![Screenshot 2024-09-30 at 23 43 44](https://github.com/user-attachments/assets/e4ef4338-020d-426c-a519-46f22220e423)


![Screenshot 2024-09-30 at 23 43 54](https://github.com/user-attachments/assets/a1b9e6bb-91b3-4b11-98bc-6163b343dd07)

@ycarogabriel  @gmendezg  @hoangsonqn

**Comment by NoSubstitute** [comment]
> https://stackoverflow.com/questions/66633659/can-not-run-trivial-app-script-bound-to-new-blank-document-get-this-app-is-bl

No, that bug/error is not related to the suggested solution posted by @santi-ifontana 
This specific solution _only_ works for Workspace accounts, where the admin has enabled _API Access Control_ to block unknown apps, perhaps also internal scripts. There's a separate setting for that here.

https://admin.google.com/ac/owl/settings?hl=en

![image](https://github.com/user-attachments/assets/bccca925-7110-418d-9606-2fd86182eb3b)

If you use the published version of ericyd's script, then this _Internal apps_ setting isn't relevant to your issues.

Then it's more likely that manually setting it as _Trusted_ will fix it.

If you are using a gmail.com-email address, then you can't use this solution.

**Comment by ycarogabriel** [comment]
> Hey everyone, same issue here. Probably you are using Google Workspace, right?
> 
> Do the following to fix it (only valid for Google Workspace accounts).
> 
> 1. Go here: https://admin.google.com/ac/owl/list?tab=configuredApps
> 2. Be sure to log in with an Admin user and the same console where the user trying to do the copy lives.
> 3. Click "Configure new app".
> 4. On the search bar that appears, type this: 983475130545-e81q4d30k615fsr942krcejaajriv23a.apps.googleusercontent.com
> 5. Copy folder should be found.
> 6. Select Trust and save.
> 7. Try again in a few minutes (sometimes it takes some time to propagate).
> 8. Good luck!
> 9. Please confirm if it worked for you!
> 
> ![Screenshot 2024-09-30 at 23 43 44](https://private-user-images.githubusercontent.com/165914151/372304031-e4ef4338-020d-426c-a519-46f22220e423.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4Nzc2OTEsIm5iZiI6MTcyNzg3NzM5MSwicGF0aCI6Ii8xNjU5MTQxNTEvMzcyMzA0MDMxLWU0ZWY0MzM4LTAyMGQtNDI2Yy1hNTE5LTQ2ZjIyMjIwZTQyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwMlQxMzU2MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Y2M4OTBjMzRiYTFjOTI5ODQ5MTczYmEyOGU5NjQ2ZjliZTFiM2JjYWIzNTkzYTYyNTc2MDdmMjFkMDJhYjg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.j2B4IACFhpKUUoGYqcamQ8aFef__u5YoAVan9d85iI0)
> 
> ![Screenshot 2024-09-30 at 23 43 54](https://private-user-images.githubusercontent.com/165914151/372304034-a1b9e6bb-91b3-4b11-98bc-6163b343dd07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4Nzc2OTEsIm5iZiI6MTcyNzg3NzM5MSwicGF0aCI6Ii8xNjU5MTQxNTEvMzcyMzA0MDM0LWExYjllNmJiLTkxYjMtNGIxMS05OGJjLTYxNjNiMzQzZGQwNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwMlQxMzU2MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NjI2NTE3NWRmM2RhY2NiMGMxZjMzYTZlYTVkYTBkYTQ2MjQ3ODQ2NTVkMjA0OWIyNDQ5MDcxNjQxY2I1YjM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.utK-tsKQVXVq6kkZrBVFU7ppA7sYqyK4TC89Kpcneqs)
> 
> @ycarogabriel @gmendezg @hoangsonqn

Worked!!! Thanks!

**Comment by wirewendy** [comment]
Hello, I use this app all the time, and can't figure out how to get it to start working again.  I do not have Google Workspace. When I click on the above links, I hit a dead end.

Any suggestions?  When I try to access it, a prompt to  "Review Permissions".  My email is presented as an option to select the correct account, but then  I end up with the message, 

"This app is blocked
This app tried to access sensitive info in your Google Account. To keep your account safe, Google blocked this access."

**Comment by santi-ifontana** [comment]
@wirewendy I have no idea what to do in your case (no Google Workspace account). As the owner of this code so far has not commented on this issue, maybe using an alternative tool can help you out. Rclone is excellent, sadly it requires much more knowledge and a VM to get it working!
